rfcs/0002-bucklescript-esy.md

```diff
@@ -0,0 +1,58 @@
+- Name: bucklescript-esy
+- Start Date: 2019-07-09
+- Issue/PR: (leave this empty)
+- Point of Contact: ulrikstrid
+
+# Summary
+
+[summary]: #summary
+
+Make Bucklescript builable with esy and make it Plug'n'Play compatible which fully unlocks esy.
+
+# Motivation
+
+[motivation]: #motivation
+
+If we can get Bucklescript we can gather the community around 1 package manager and we might also be able to run basically any ppx by just depending on the correct OCaml compiler. We should also be able to create mono-repos with Bucklescript for the frontend code and native for backend.
```

Comment:
So I read that some of the motivation is that:

1. The community is split across package managers (yarn, esy, opam)
2. We are not able to run arbitrary PPXs with arbitrary compiler versions
3. We can't create mono repos easily that have both web modules and native modules

We should rephrase this to be clear about:

* what are the current blockers with BuckleScript (bsc + bsb requiring node)
* what could we enable by doing this work (e.g, multitarget monorepos, single dev flow for all targets, enable pnp support for bsc, etc)